src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -916,4 +948,28 @@ String n(Run<?, ?> b, String label) {
         }
     }
 
+    @TestExtension("queueTaskOwnerCorrectWhenRestarting")
+    public static class PipelineOnlyTaskDispatcher extends QueueTaskDispatcher {
+        @Override
+        public CauseOfBlockage canTake(Node node, Queue.BuildableItem item) {
+            final Queue.Task[] t = new Queue.Task[]{item.task};
```

Comment:
Clearer to either use a `final AtomicReference<Queue.Task>`, or just a (non`final`) `Queue.Task` and assign to a `final` variable before creating the anonymous inner class.